lib/generators/administrate/routes/routes_generator.rb

```diff
@@ -8,7 +8,7 @@ class RoutesGenerator < Rails::Generators::Base
       source_root File.expand_path("../templates", __FILE__)
 
       def insert_dashboard_routes
-        unless File.read(rails_routes_file_path).include?(dashboard_routes)
+        unless routes_includes_resources? || valid_dashboard_models.empty?
```

Comment:
Do you mind moving these conditions into a `should_route_dashboard?` method? I struggle to read `unless`, especially with a longer logical expression.